proposals/text/5289-recency-reranker.md

```diff
@@ -0,0 +1,164 @@
+- Title: Addition of a 'RecencyReranker' node
+- Decision driver: @elundaeva
+- Start Date: 2023-07-05
+- Proposal PR: https://github.com/deepset-ai/haystack/pull/5289
+- Github Issue or Discussion: some links available in the "Detailed design" section below
+
+# Summary
+
+This reranker allows to have retrieved documents sorted not only by relevance (default) but also with recency factored in.
+
+# Basic example
+
+The RecentnessReranker would be instantiated as follows:
+
+  ``` python
+
+	  reranker = RecentnessReranker(
```

Comment:
I would go ahead and update all instances of `RecentnessReranker` and `RecencyReranker` to `RecentnessRanker`